src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/Platform/SqlBulkBatchWriter.cs

```diff
@@ -17,19 +18,52 @@ internal class SqlBulkBatchWriter : ISqlBulkBatchWriter
         private readonly bool _disableTriggers;
         private readonly ISqlConnectionFactory _sqlConnectionFactory;
         private readonly ILogEventDataGenerator _logEventDataGenerator;
-
+        private readonly ColumnOptions _columnOptions;
+        private readonly TimeSpan? _retentionPeriod;
+        private readonly TimeSpan? _cleaningInterval;
+        private long _lastCleaning;
         public SqlBulkBatchWriter(
+
             string tableName,
             string schemaName,
             bool disableTriggers,
             ISqlConnectionFactory sqlConnectionFactory,
-            ILogEventDataGenerator logEventDataGenerator)
+            ILogEventDataGenerator logEventDataGenerator,
+             ColumnOptions columnOptions = null,
+             MSSqlServerSinkOptions MSSqlServerSinkOptions = null
+            )
         {
             _tableName = tableName ?? throw new ArgumentNullException(nameof(tableName));
             _schemaName = schemaName ?? throw new ArgumentNullException(nameof(schemaName));
             _disableTriggers = disableTriggers;
             _sqlConnectionFactory = sqlConnectionFactory ?? throw new ArgumentNullException(nameof(sqlConnectionFactory));
             _logEventDataGenerator = logEventDataGenerator ?? throw new ArgumentNullException(nameof(logEventDataGenerator));
+            _columnOptions = columnOptions;
+            _retentionPeriod = MSSqlServerSinkOptions?.RetentionPeriod ;
+            _cleaningInterval = MSSqlServerSinkOptions?.CleaningInterval;
+            _lastCleaning = DateTime.Now.Ticks;
+        }
+
+        public async Task cleanOldLogEvents()
+        {
+            //Preventing other threads from unnecessary cleaning 
+            var lastCleaning_ticks = Interlocked.Exchange(ref _lastCleaning, DateTime.Now.Ticks);
```

Comment:
_lastCleaning is also updated when an exception happens. Would this cause an invervall (e.g. 2 hours) to be skipped?

Reply:
Exception handled and wouldn't be skipped anymore.